controllers/purge/suite_test.go

```diff
@@ -0,0 +1,148 @@
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	//+kubebuilder:scaffold:imports
+)
+
+// These tests use Ginkgo (BDD-style Go testing framework). Refer to
+// http://onsi.github.io/ginkgo/ to learn more about Ginkgo.
+
+const UseRandomPort = "0"
+
+var (
+	purgeReconciler    *controllers.PurgeReconciler //nolint:gochecknoglobals
+	controlPlaneClient client.Client                //nolint:gochecknoglobals
+	k8sManager         manager.Manager              //nolint:gochecknoglobals
+	singleClusterEnv   *envtest.Environment         //nolint:gochecknoglobals
+	ctx                context.Context              //nolint:gochecknoglobals
+	cancel             context.CancelFunc           //nolint:gochecknoglobals
+	cfg                *rest.Config                 //nolint:gochecknoglobals
+)
+
+func TestAPIs(t *testing.T) {
+	t.Parallel()
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Purge Controller Suite")
+}
+
+var _ = BeforeSuite(func() {
+	ctx, cancel = context.WithCancel(context.TODO())
+	logf.SetLogger(log.ConfigLogger(9, zapcore.AddSync(GinkgoWriter)))
+
+	By("bootstrapping test environment")
+
+	externalCRDs := AppendExternalCRDs(
+		filepath.Join("..", "..", "config", "samples", "tests", "crds"),
+		"cert-manager-v1.10.1.crds.yaml",
+		"istio-v1.17.1.crds.yaml")
+
+	singleClusterEnv = &envtest.Environment{
+		CRDDirectoryPaths:     []string{filepath.Join("..", "..", "config", "crd", "bases")},
+		CRDs:                  append([]*apiExtensionsv1.CustomResourceDefinition{}, externalCRDs...),
+		ErrorIfCRDPathMissing: true,
+	}
+
+	cfg, err := singleClusterEnv.Start()
+	Expect(err).NotTo(HaveOccurred())
+	Expect(cfg).NotTo(BeNil())
+
+	Expect(api.AddToScheme(scheme.Scheme)).NotTo(HaveOccurred())
+	Expect(apiExtensionsv1.AddToScheme(scheme.Scheme)).NotTo(HaveOccurred())
+
+	//+kubebuilder:scaffold:scheme
+
+	k8sManager, err := ctrl.NewManager(
+		cfg, ctrl.Options{
+			MetricsBindAddress: UseRandomPort,
+			Scheme:             scheme.Scheme,
+			NewCache:           controllers.NewCacheFunc(),
+		})
+	Expect(err).ToNot(HaveOccurred())
+
+	intervals := controllers.RequeueIntervals{
+		Success: 3 * time.Second,
+	}
+
+	remoteClientCache := remote.NewClientCache()
+
+	var useLocalClient controllers.RemoteClientResolver = func(context.Context, client.ObjectKey) (client.Client, error) {
+		return k8sManager.GetClient(), nil
+	}
+
+	purgeReconciler = &controllers.PurgeReconciler{
+		Client:           k8sManager.GetClient(),
+		EventRecorder:    k8sManager.GetEventRecorderFor(operatorv1beta1.OperatorName),
+		RequeueIntervals: intervals,
+		VerificationSettings: signature.VerificationSettings{
+			EnableVerification: false,
+		},
+		RemoteClientCache:     remoteClientCache,
+		ResolveRemoteClient:   useLocalClient,
+		KcpRestConfig:         k8sManager.GetConfig(),
+		PurgeFinalizerTimeout: time.Second, //TODO: As short as possible?
```

Comment:
Anything TODO left?

Reply:
We want the timeout to be as short as possible for the testing in the pipelines, else it can take unusually long to just build. 1 Second is what we came down to. But in case the tests fail due to timeout issues in the future, then this timeout can be increased.